src/mongoose_client_api_messages.erl

```diff
@@ -106,12 +106,33 @@ make_json_msg(Msg, MAMId) ->
 
 -spec encode(exml:item(), integer()) -> map().
 encode(Msg, Timestamp) ->
+    RawProps = exml_query:path(Msg, [{element,<<"properties">>}]),
     BodyTag = exml_query:path(Msg, [{element, <<"body">>}]),
-    #{from => exml_query:attr(Msg, <<"from">>),
-      to => exml_query:attr(Msg, <<"to">>),
-      id => exml_query:attr(Msg, <<"id">>),
-      body => exml_query:cdata(BodyTag),
-      timestamp => Timestamp}.
+
+    CoreList = [{<<"from">>, exml_query:attr(Msg, <<"from">>)},
+                {<<"to">>, exml_query:attr(Msg, <<"to">>)},
+                {<<"id">>,exml_query:attr(Msg, <<"id">>)},
+                {<<"body">>,exml_query:cdata(BodyTag)},
+                {<<"timestamp">>,Timestamp}],
+
+    ExtensionList =
+      case RawProps of
+           #xmlel{name = Name
+              ,attrs = Attrs
+              ,children = Children} ->
+           Props = [convert_prop_child(Child) || Child <- Children],
+           [{<<"properties">>,maps:from_list(Props)}];
+       _ -> []
+      end,
+
+    maps:from_list(CoreList ++ ExtensionList).
```

Comment:
It's more efficient to create `ExtensionList` first, then create full list with `[{<<"from">>,... | ExtensionList]`